### PR TITLE
Solving missing line param on gitlab codequality

### PR DESCRIPTION
**Pull request by slipknois**
A compliant report need to output lines on "location" object to work properly as doc in link show us: https://docs.gitlab.com/ci/testing/code_quality/

This commit write de "lines" property properly. If "location" object don't output the line, the gitlab show a message saying "Failed to load Code Quality report"

Issue: https://github.com/google/yamlfmt/issues/272

### CODE REVIEW

**Comment by braydonk** [comment]
This PR currently undoes some of my requested changes on #294. Please go back through those and ensure they are all done.

In addition, I can tell just by looking that the given test will fail. Please run the tests and fix the failure.

**Comment by braydonk** [comment]
I still don't see evidence that the author of this PR actually ran the tests (I know they will fail by reading). In addition, old comments I made on the PR were undone in this reopen.

File: internal/gitlab/codequality.go [hunk 1]
```diff
@@ -46,17 +54,66 @@ func NewCodeQuality(diff yamlfmt.FileDiff) (CodeQuality, bool) {
...
+			end = i + 1
+		}
+	}
+
+	// fallback (should not happen because diff.Changed() was true)
+	if begin == -1 {
+		begin = 1
+	}
+	if end == -1 {
+		end = 1
```
Comment (Hazer) [inline internal/gitlab/codequality.go]: Can't we omit this if end is omitempty?

**Comment by slipknois** [comment]
@braydonk can you review again please?

**Comment by braydonk** [comment]
2 things:

1. Local branch still isn't rebased.
2. How did you do the stdout.txt update? Did you use `make integrationtest_update`? If not, please do it with that (the diff might end up being the same)

**Comment by slipknois** [comment]
> 2. How did you do the stdout.txt update? Did you use `make integrationtest_update`? If not, please do it with that (the diff might end up being the same)

I used `make integrationtest_update`

File: internal/gitlab/codequality.go [hunk 2]
```diff
@@ -46,17 +54,63 @@ func NewCodeQuality(diff yamlfmt.FileDiff) (CodeQuality, bool) {
...
 		},
 	}, true
 }
 
+// detectChangedLines finds the first and last lines that differ between original and formatted content.
+func detectChangedLines(diff *yamlfmt.FileDiff) (begin int, end *int) {
+	original := strings.Split(string(diff.Diff.Original), "\n")
+	formatted := strings.Split(string(diff.Diff.Formatted), "\n")
+
+	max := len(original)
```
Comment (braydonk) [inline internal/gitlab/codequality.go]: This can just use `math.Max` instead.

File: internal/gitlab/codequality_test.go [hunk 3]
```diff
@@ -93,3 +95,186 @@ func TestCodeQuality(t *testing.T) {
 		})
 	}
 }
+
+func TestCodeQuality_DetectChangedLine(t *testing.T) {
```
Comment (braydonk) [inline internal/gitlab/codequality_test.go]: This should be a table test with more test cases. The `testdata` directory should have multiple different examples of changed lines. Some edge cases I would like to see:
* No lines changed
* All lines changed
* One line changed
* Only final line changed
* Only change is one final line added

File: internal/gitlab/codequality.go (hunk 2 above)
Comment (braydonk) [reply internal/gitlab/codequality.go]: Right, perhaps we can use the [`max` builtin`](https://pkg.go.dev/builtin#max) then.

File: internal/gitlab/testdata/changed_line/all_lines_changed/formatted.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+x: y
+z: w
```
Comment (braydonk) [inline internal/gitlab/testdata/changed_line/all_lines_changed/formatted.yaml]: In a realistic scenario, the two things that are being compared for codequality will both end with new lines. All the testdata files should have new lines at the end of the files.

**Review by braydonk — COMMENTED** [review]
Almost there, one new comment and one comment remains unaddressed.

**Comment by slipknois** [comment]
@braydonk can you review again please?

**Review by braydonk — APPROVED** [review]
